bedms/train.py

```diff
@@ -0,0 +1,242 @@
+            logger.error(
+                f"Mismatch in number of value files ({len(values_files_list)}) \
+                and header files ({len(headers_files_list)})"
+            )
+            return
+
+        total_files = len(values_files_list)
+
+        paired_files = list(zip(values_files_list, headers_files_list))
+
+        train_size = self.config["data_split"]["train_set"]
+        test_size = self.config["data_split"]["test_set"]
+        val_size = self.config["data_split"]["val_set"]
+
+        if train_size + val_size + test_size > total_files:
+            logger.error(
+                f"Data split sizes exceed total number of files: "
+                f"train({train_size}) + val({val_size}) + \
+                test({test_size}) > total_files({total_files})"
+            )
+            return
+
+        train_files = paired_files[:train_size]
+        val_files = paired_files[train_size : train_size + val_size]
+        test_files = paired_files[
+            train_size + val_size : train_size + val_size + test_size
+        ]
+
+        logger.info(f"Training on {len(train_files)} file sets")
+        logger.info(f"Validating on {len(val_files)} file sets")
+        logger.info(f"Testing on {len(test_files)} file sets")
+
+        x_values_train_list, x_headers_train_list, y_train_list = accumulate_data(
+            train_files
+        )
+        x_values_test_list, x_headers_test_list, y_test_list = accumulate_data(
+            test_files
+        )
+        x_values_val_list, x_headers_val_list, y_val_list = accumulate_data(val_files)
+
+        logger.info("Accumulation Done.")
+
+        num_cluster = self.config["training"]["num_cluster"]
+        vectorizer_pth = self.config["training"]["vectorizer_pth"]
+        label_encoder_pth = self.config["training"]["label_encoder_pth"]
+        sentence_transformer_model = self.config["training"][
+            "sentence_transformer_model"
+        ]
+
+        (
+            train_encoded_data,
+            test_encoded_data,
+            val_encoded_data,
+            self.label_encoder,
+            self.vectorizer,
+        ) = training_encoding(
+            x_values_train_list,
+            x_headers_train_list,
+            y_train_list,
+            x_values_test_list,
+            x_headers_test_list,
+            y_test_list,
+            x_values_val_list,
+            x_headers_val_list,
+            y_val_list,
+            num_cluster,
+            vectorizer_pth,
+            label_encoder_pth,
+            sentence_transformer_model,
+        )
+        logger.info("Encoding Done.")
+
+        batch_size = self.config["training"]["batch_size"]
+        self.train_loader = data_loader(train_encoded_data, batch_size)
+        self.test_loader = data_loader(test_encoded_data, batch_size)
+        self.val_loader = data_loader(val_encoded_data, batch_size)
+
+        logger.info("Loading Done.")
+
+    def training(self):
```

Comment:
As mentioned in the README, I prefer `.train()` here. Feels more like a call to action, where `.training()` almost feels like i am checking or setting the status of something